storage/indexer/src/db_indexer.rs

```diff
@@ -273,13 +290,25 @@ impl InternalIndexerDB {
             .get::<InternalIndexerMetadataSchema>(key)?
             .map(|v| v.expect_version()))
     }
+
+    pub fn get_translated_v1_event_by_version_and_index(
+        &self,
+        version: Version,
+        index: u64,
+    ) -> Result<ContractEventV1> {
+        self.db
+            .get::<TranslatedV1EventSchema>(&(version, index))?
+            .ok_or_else(|| AptosDbError::NotFound(format!("Event {} of Txn {}", index, version)))
+    }
 }
 
 pub struct DBIndexer {
     pub indexer_db: InternalIndexerDB,
     pub main_db_reader: Arc<dyn DbReader>,
     sender: Sender<Option<SchemaBatch>>,
     committer_handle: Option<thread::JoinHandle<()>>,
+    event_sequence_number_cache: Mutex<HashMap<EventKey, u64>>,
```

Comment:
can we put this cache into translators?

Reply:
I moved it into `EventV2TranslationEngine`.